app/uk/gov/hmrc/gform/validation/ComponentValidator.scala

```diff
@@ -251,6 +252,17 @@ object ComponentValidator {
     sharedTextComponentValidator(fieldValue, value, min, max, ValidText, messageKey)
   }
 
+  private def referenceNumberConstraints(fieldValue: FormComponent, value: String, min: Int, max: Int)(
+    implicit
+    messages: Messages,
+    sse: SmartStringEvaluator
+  ) = {
+    val ValidReferenceNumber = s"[0-9]{$min,$max}".r
+    val str = value.replace(" ", "")
+    val messageKey = "generic.referenceNumber.error.pattern"
```

Comment:
would it be useful to also include min and max length allowed, in the error message?

Reply:
Min/Max length is dealt with in the `sharedTextComponentValidator` method.